src/my-requests.tsx

```diff
@@ -208,114 +210,180 @@ const ProceedingRow = (props: ProceedingRowProps) => {
 
     const original_request = findOriginalRequest(props.proceeding);
 
+    const initialMessage = useMemo<Omit<Message, 'id'>>(
+        () => ({
+            transport_medium: 'email',
+            date: new Date(),
+            reference: props.proceeding.reference,
+            sentByMe: false,
+            correspondent_address: original_request?.correspondent_address || '',
+            correspondent_email: original_request?.correspondent_email || '',
+            type: 'response',
+        }),
+        [props.proceeding.reference, original_request]
+    );
+
+    const ImportModalContent = () => {
+        const [newMessage, setNewMessage] = useState<Omit<Message, 'id'>>(initialMessage);
+        const addMessage = useProceedingsStore((state) => state.addMessage);
+
+        return (
+            <>
+                <p>
+                    <Text id="import-message-modal-explanation" />
+                </p>
+                <MessageMetadataInput
+                    message={newMessage}
+                    onChange={(message) => setNewMessage(message)}
+                    includeContent={true}
+                />
+                <button
+                    className="button button-secondary"
+                    style="margin-top: 10px;"
+                    onClick={() => {
+                        setNewMessage(initialMessage);
+                        dismissImportMessageModal();
+                    }}>
+                    <Text id="cancel" />
+                </button>
+                <button
+                    className="button button-primary"
+                    style="margin-left: 5px; margin-top: 10px;"
+                    onClick={() => {
+                        addMessage(newMessage);
+                        setNewMessage(initialMessage);
+                        dismissImportMessageModal();
+                    }}>
+                    <Text id="add-message" />
+                </button>
+            </>
+        );
+    };
+
+    const [ImportMessageModal, showImportMessageModal, dismissImportMessageModal] = useModal(<ImportModalContent />, {
+        defaultButton: 'positive',
+        backdropDismisses: true,
+        escDismisses: true,
+        hasDismissButton: true,
```

Comment:
I don't think we can guard those with a `confirm()`, so I think we should disable them.